crates/compilers/src/cache.rs

```diff
@@ -668,10 +686,17 @@ impl<T: ArtifactOutput<CompilerContract = C::CompilerContract>, C: Compiler>
             .map(|import| strip_prefix(import, self.project.root()).into())
             .collect();
 
+        let interface_repr_hash = if self.cache.preprocessed {
+            self.is_source_file(&file).then(|| interface_representation_hash(source, &file))
```

Comment:
can we calculate this somehow when computing SolData? Currently we parse each file 3 times:
1. for SolData
2. for interface repr hash
3. for preprocessing

Parsing is pretty fast but I would like to avoid re-parsing same files multiple times

Reply:
would it work if we compute and add it in `Source` struct as an option when we read https://github.com/foundry-rs/compilers/blob/33b3ca8b15d5edecfdeef7e47e3757476a263c49/crates/artifacts/solc/src/sources.rs#L127-L138

Then on `create_cache_entry` we just add it in the cache, wdyt?

Reply:
Pushed https://github.com/foundry-rs/compilers/pull/252/commits/59ebda6a2111b27f352a414849934715151d2e8c

Reply:
actually realized that SolData doesn't get parsed in parallel which makes it overall slower

Reply:
got it, makes sense to compute `interface_repr_hash` in `SolData.parse` where we already have the ast

Reply:
Full diff https://github.com/foundry-rs/compilers/pull/252/files/fb5ebc251f58727d364e5c58d330be14797b73f6..4451c71fbf2fc552002c4a4b90ecc5e352fb431b

Reply:
looks good, mind to share what was the perf difference you seen between creating the interface hash in `SolData.parse` vs in artifact cache `fill_hashes`?

Reply:
slightly faster overall, I was wrong about sequential vs parallel since we always fill the hashes sequentially which is kind of unfortunate